cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedTable.java

```diff
@@ -59,13 +60,14 @@
  * </p>
  *
  * <p>The indexed values need not be unique.  When reading the data back by index value, a {@link Scanner} will be
- * returned, allowing the client to iterate through all matching rows. Only exact matches on indexed values
- * are currently supported.
+ * returned, allowing the client to iterate through all matching rows. Exact matches as well as range lookups on
+ * indexed values are supported.
  * </p>
  *
  * <p>Index entries are created by storing additional rows in a second table.  These index rows are keyed by
  * column name, column value, and original row key, each field separated by a single null byte delimiter.  Values
- * containing the null byte are allowed, but may be read and filtered out when reading from the index.
```

Comment:
Thinking about this more I think the wording is not that clear. Perhaps this should just say: It is not recommended that the name of an index column contains the null byte (this can cause a degradation in performance of index reads).